gitpod-network-check/cmd/checks.go

```diff
@@ -198,13 +209,22 @@ func checkSMPrerequisites(ctx context.Context, ec2Client *ec2.Client) error {
 		}
 
 		if len(response.VpcEndpoints) == 0 {
-			if endpoint.Required {
-				return fmt.Errorf("❌ VPC endpoint %s not configured: %w", endpoint.Endpoint, err)
+			if strings.Contains(endpoint.Endpoint, "execute-api") {
+				log.Infof("ℹ️  Deferring connectivity test for %s service until testing main subnet", endpoint.PrivateDnsName)
+				continue
+			}
+			log.Infof("ℹ️  VPC endpoint %s is not configured, testing service connectivity...", endpoint.Endpoint)
+			_, err := TestServiceConnectivity(ctx, endpoint.PrivateDnsName, 5*time.Second)
```

Comment:
This new method will help us assert whether or not we have connectivity to VPC endpoints which are necessary to send SSM commands to EC2 instances (which later asserts connectivity, by subnet, to AWS services).